### PR TITLE
feat: add ratelimit information

### PR DESCRIPTION
**Pull request by peterdeme**
Add rate limit information to the response

### CODE REVIEW

File: stream_chat/base/client.py [hunk 1]
```diff
@@ -26,6 +28,24 @@ def __init__(
...
         )
 
+    def _enrich_response(
+        self,
+        response: Union[aiohttp.ClientResponse, requests.Response],
+        resp_body: StreamResponse,
+    ) -> StreamResponse:
+        resp_headers: Dict[str, Any] = dict(response.headers)  # type: ignore
+
+        resp_body["rate_limit"] = dict(
```
Comment (gumuz) [inline stream_chat/base/client.py:38]: how sure are we these headers are always present?

Comment (peterdeme) [reply stream_chat/base/client.py]: @gumuz refactored it

**Review by tbarbugli — CHANGES_REQUESTED** [review]
1. we should come up with a solution that works for all languages and is as consistent as possible
2. the approach in this PR adds data to the response `dict` this is potentially leaking data to objects in a way that customers do not expect

**Comment by peterdeme** [comment]
@tbarbugli thanks for the feedback.

> we should come up with a solution that works for all languages and is as consistent as possible

Agreed. Btw, I copied this approach from the Java SDK: https://github.com/GetStream/stream-chat-java/blob/c2897959a13d05f693e00ed47c4a8cc2baaf66c3/src/main/java/io/getstream/chat/java/services/framework/StreamServiceHandler.java#L52-#L63

> the approach in this PR adds data to the response dict this is potentially leaking data to objects in a way that customers do not expect

Yeah, that's a concern.

As for ideas. I have an idea but it's a breaking change:
```py
from typing import TypeVar
from dataclasses import dataclass

T = TypeVar("T")

class RateLimit(dataclass):
  pass

class StreamResponse(dataclass):
  body: T
  rate_limit: RateLimit

def get_something_from_backend() -> StreamResponse:
   var response = request() # request here
   return StreamResponse(response.body.json(), get_ratelimit_info(response))

**Comment by ferhatelmas** [comment]
> As for ideas. I have an idea but it's a breaking change:

1. This idea sounds good to me. Similar to what go sdk does too.
2. Breaking change is fine, we can bump a major.

**Comment by ferhatelmas** [comment]
we can drop 3.6 in next major since it became eol last week.

**Comment by peterdeme** [comment]
Ok, I'll create a new PR.

**Pull request closed without merging**